tools/update-rust.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+set -e
+
+if [[ $# -lt 1 ]]; then
+    echo "Usage: $0 $RUST_VERSION"
+    echo
+    echo "Example: $0 1.71.0"
+    exit 2
+fi
+
+TOOLSDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+ROOTDIR=$(realpath "${TOOLSDIR}/..")
+
+VERSION="${1}"
+OUTPUT="${ROOTDIR}/hashes/rust"
+PACKAGE_ROOT="https://static.rust-lang.org"
+METADATA_FILE="metadata.json"
+ARCHES=("x86_64" "aarch64")
+PACKAGES=("rust-std" "rustc" "cargo")
+
+function get_package_url {
+    local package="$1"
+    local arch="$2"
+
+    path=$(grep -e "${package}.*${arch}-unknown-linux-gnu.tar.xz" "${METADATA_FILE}" | cut -d '"' -f 3)
+    path=$(echo "${path//\\}")
+
+    echo "${path}"
+}
+
+function add_package {
+    local package="$1"
+    local arch="$2"
+
+    PKG_PATH=$(get_package_url "${package}" "${arch}")
+    PKG_FILE=$(echo "${PKG_PATH}" | cut -d '/' -f 3)
+    wget "${PACKAGE_ROOT}/${PKG_PATH}"
+    PKG_SHA=$(sha512sum "${PKG_FILE}" | cut -d ' ' -f 1)
+    rm "${PKG_FILE}"
+
+    echo "# ${PACKAGE_ROOT}/${PKG_PATH}" >> "${OUTPUT}"
+    echo "SHA512 (${PKG_FILE}) = ${PKG_SHA}" >> "${OUTPUT}"
+}
+
+cd /tmp
+
+# Get the rustc source package
+RUSTC_PACKAGE="rustc-${VERSION}-src.tar.xz"
+RUSTC_URL="${PACKAGE_ROOT}/dist/${RUSTC_PACKAGE}"
+
+wget "${RUSTC_URL}"
```

Comment:
I'm admittedly a `wget` fan, but I might recommend using `curl` instead since it's more likely to be installed on everyone's machine.

Reply:
Switched